src/sphinxexpr.cpp

```diff
@@ -5908,6 +6089,15 @@ ISphExpr * ExprParser_t::CreateUdfNode ( int iCall, ISphExpr * pLeft )
 		case SPH_ATTR_STRINGPTR:
 			pRes = new Expr_UdfStringptr_c ( m_dUdfCalls[iCall], m_pProfiler );
 			break;
+		case SPH_ATTR_UINT32SET_PTR:
+			pRes = new Expr_UdfMva32_c ( m_dUdfCalls[iCall], m_pProfiler );
```

Comment:
Expr_UdfMva32_c  and Expr_UdfMva64_c is the full clone of each other. The only difference are multiple asserts messages
```
- assert ( 0 && "internal error: mva64 udf evaluated as float" );
+ assert ( 0 && "internal error: mva32 udf evaluated as float" );
```

and also this assert
```
- assert ( pCall->m_pUdf->m_eRetType==SPH_ATTR_UINT32SET_PTR );
+ assert ( pCall->m_pUdf->m_eRetType==SPH_ATTR_INT64SET_PTR );
```

I sure it could be the single `Expr_UdfMva_c` class that could has `bool m_bMva64` set at the constructor then that bool could control the all asserts messages or behavior for assert for the type check like
```
assert ( ( !m_bMva64 && pCall->m_pUdf->m_eRetType==SPH_ATTR_UINT32SET_PTR ) || ( m_bMva64 && pCall->m_pUdf->m_eRetType==SPH_ATTR_INT64SET_PTR ) );
```